kernel/src/engine/default/json.rs

```diff
@@ -2,19 +2,22 @@
 
 use std::io::BufReader;
 use std::ops::Range;
-use std::sync::Arc;
-use std::task::{ready, Poll};
+use std::sync::{mpsc, Arc};
```

Comment:
Consider [`tokio::sync::mpsc`](https://docs.rs/tokio/latest/tokio/sync/mpsc/fn.channel.html) instead? much faster, designed to be used in async context

Reply:
we don't depend on tokio in the default except for implementing executors in terms of it. it might be fine, but for now we can stay stdlib